core/arch/aarch64/instr.c

```diff
@@ -53,6 +53,8 @@ instr_length_arch(dcontext_t *dcontext, instr_t *instr)
 {
     if (instr_get_opcode(instr) == OP_LABEL)
         return 0;
+    if (instr_get_opcode(instr) == OP_ldstex)
+        return opnd_get_immed_int(instr_get_src(instr, 0)) * AARCH64_INSTR_SIZE;
```

Comment:
Correction: not done. It turns out not to be straightforward. I think there is architecture-independent code somewhere that sets length to zero so the value gets lost if I don't hide it in a pseudo-operand.